artemis-boot/src/main/java/org/apache/activemq/artemis/boot/Artemis.java

```diff
@@ -112,7 +112,7 @@ public int compare(File file, File file1) {
          }
       }
 
-      if (fileInstance != null) {
+      if (System.getProperty("java.io.tmpdir") == null) {
```

Comment:
I think the `fileInstance != null` should still be there as it's still valid.

Reply:
Ah, yes, you're right. Updated.